### PR TITLE
[Flan-UL2] Add-flan-ul2

### LINKED ISSUE #21917: Add FLAN-UL2
### Model description

UL2 is a unified framework for pretraining models that are universally effective across datasets and setups. UL2 uses Mixture-of-Denoisers (MoD), a pre-training objective that combines diverse pre-training paradigms together. UL2 introduces a notion of mode switching, wherein downstream fine-tuning is associated with specific pre-training schemes.

FLAN-UL2 has the same configuration as the original UL2 20B model, except that it has been instruction tuned with Flan. 

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

The model architecture (UL2) is already in Huggingface Transformers.
The 20B model weights are here: https://github.com/google-research/google-research/tree/master/ul2#checkpoints

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Adds the documentation for the FLan-UL2 model
cc @younesbelkada 
Fixes #21917

### CODE REVIEW

File: docs/source/en/flan-ul2.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+>>> from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
+
+>>> model = AutoModelForSeq2SeqLM.from_pretrained("google/flan-ul2")
+>>> tokenizer = AutoTokenizer.from_pretrained("google/flan-ul2")
+
+>>> inputs = tokenizer("A step by step recipe to make bolognese pasta:", return_tensors="pt")
+>>> outputs = model.generate(**inputs)
+>>> print(tokenizer.batch_decode(outputs, skip_special_tokens=True))
+['Pour a cup of bolognese into a large bowl and add the pasta']
+```
```
Comment (younesbelkada) [inline docs/source/en/flan-ul2.mdx:29]: This takes too much memory as it'll run on CPU, can you add more details on how to leverage `accelerate` and/or `bitsandbytes`?

Comment (younesbelkada) [inline docs/source/en/flan-ul2.mdx:29]:
```suggestion

### Running the model on CPU

If you have enough CPU RAM to run the model in `bfloat16`, you can run the following snippet to execute the model on CPU:
```python
>>> from transformers import AutoModelForSeq2SeqLM, AutoTokenizer

>>> model = AutoModelForSeq2SeqLM.from_pretrained("google/flan-ul2")
>>> tokenizer = AutoTokenizer.from_pretrained("google/flan-ul2")

>>> inputs = tokenizer("A step by step recipe to make bolognese pasta:", return_tensors="pt")
>>> outputs = model.generate(**inputs)
>>> print(tokenizer.batch_decode(outputs, skip_special_tokens=True))
['Pour a cup of bolognese into a large bowl and add the pasta']
```

### Running the model on GPU

...
```

File: README.md [hunk 2]
```diff
@@ -329,6 +329,7 @@ Current number of checkpoints: ![](https://img.shields.io/endpoint?url=https://h
...
+1. **[FLAN-UL2](https://huggingface.co/docs/transformers/model_doc/flan-ul2)** (from Google AI) released in the repository [google-research/t5x](https://github.com/google-research/t5x/blob/main/docs/models.md#flan-ul2-checkpoints) by Hyung Won Chung, Le Hou, Shayne Longpre, Barret Zoph, Yi Tay, William Fedus, Eric Li, Xuezhi Wang, Mostafa Dehghani, Siddhartha Brahma, Albert Webson, Shixiang Shane Gu, Zhuyun Dai, Mirac Suzgun, Xinyun Chen, Aakanksha Chowdhery, Sharan Narang, Gaurav Mishra, Adams Yu, Vincent Zhao, Yanping Huang, Andrew Dai, Hongkun Yu, Slav Petrov, Ed H. Chi, Jeff Dean, Jacob Devlin, Adam Roberts, Denny Zhou, Quoc V. Le, and Jason Wei
```
Comment (younesbelkada) [inline README.md:332]:
```suggestion
1. **[FLAN-UL2](https://huggingface.co/docs/transformers/model_doc/flan-ul2)** (from Google AI) released in the repository [google-research/t5x](https://github.com/google-research/t5x/blob/main/docs/models.md#flan-ul2-checkpoints) by Yi Tay
```
I think Yi Tay is the only author of this model?

Comment (shermansiu) [reply README.md:332]: Tay and Dehghani are the co-leads of UL2. From the FLAN-UL2 blog post: "This work was also made possible with the help and contributions from all the authors on the UL2 paper and Flan2 paper."

**Comment by shermansiu** [comment]
Also, why are the international README entries in the wrong language?

**Comment by ArthurZucker** [comment]
This should not be the case! Will fix this.

File: README.md (hunk 2 above)
Comment (younesbelkada) [reply README.md:332]: Thanks!  Let's keep it as it is then

File: docs/source/en/model_doc/flan-ul2.mdx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+
+One can refer to [T5's documentation page](t5) for all tips, code examples and notebooks. As well as the FLAN-T5 model card for more details regarding training and evaluation of the model.
+
+The original checkpoints can be found [here](https://github.com/google-research/t5x/blob/main/docs/models.md#flan-ul2-checkpoints).
+
+
+## Running on low ressources
+The model is pretty heavy (~20GB) so if you just want to run the model, make sure you load in 8bits, and use `device_map="auto"` to make sure 
+you don't have OOM issue!
```
Comment (younesbelkada) [inline docs/source/en/model_doc/flan-ul2.mdx:36]:
```suggestion
## Running on low ressources devices

The model is pretty heavy (~40GB in half precision) so if you just want to run the model, make sure you load your model in 8bit, and use `device_map="auto"` to make sure  you don't have any OOM issue!
```

File: docs/source/en/model_doc/flan-ul2.mdx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+>>> from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
+
+>>> model = AutoModelForSeq2SeqLM.from_pretrained("google/flan-ul2", load_in_8bit=True, device_map="auto")
+>>> tokenizer = AutoTokenizer.from_pretrained("google/flan-ul2")
+
+>>> inputs = tokenizer("A step by step recipe to make bolognese pasta:", return_tensors="pt")
+>>> outputs = model.generate(**inputs)
+>>> print(tokenizer.batch_decode(outputs, skip_special_tokens=True))
+['In a large skillet, brown the ground beef and onion over medium heat. Add the garlic']
+```
```
Comment (younesbelkada) [inline docs/source/en/model_doc/flan-ul2.mdx:61]:
```suggestion
```

**Comment by younesbelkada** [comment]
@sgugger it seems that the README translations are not correct for several languages (korean, jp, etc), I think that we can address a solution in a follow-up PR!

**Comment by shermansiu** [comment]
Btw, I opened a new issue for the README translations to keep things nice and organized (#21932).

**Review by sgugger — APPROVED** [review]
Thanks for adding this. Don't worry about the translated READMEs, maybe someone in the community will fix the translations later on, but this is not high priority.

**Pull request merged**